QueryFilter/QueryFilter.php

```diff
@@ -124,11 +126,14 @@ private function getSimpleSearchBy(array $allowedCols, ?array $search): array
         }
 
         foreach ($search as $key => $val) {
-            if (!in_array($key, $allowedCols, true) || $val === null) {
+            if (in_array($key, $allowedCols, true) && $val !== null) {
```

Comment:
Thank you. I just noticed a small inconsistency of the approach. Here you inverted the condition and moved the `throw` call below. But in `getFullSearchBy`. You just added an additional condition inside the existing condition. I don't mind against any of the approaches, but it's better to have it consistent.

Also, did you think of the sort condition? Do we want to have it handled by the new feature as well?

Reply:
I rewrote them to match. Is it okay right now? I do not prefer having nested if statements if not necessarily.

Reply:
Good point. I also added check for column/type in `getSortData` method in `QueryFilter` class.

Reply:
I liked your changes. Thank you! There are a few minor issues left, and we are done. 👍